test/splitOnRegex.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+var jsc = require('jsverify');
+
+var S = require('..');
+
+var eq = require('./internal/eq');
+var equals = require('./internal/equals');
+
+
+test('splitOnRegex', function() {
+
+  eq(typeof S.splitOnRegex, 'function');
+  eq(S.splitOnRegex.length, 2);
+  eq(S.splitOnRegex.toString(), 'splitOnRegex :: RegExp -> String -> Array String');
+
+  eq(S.splitOnRegex(/b/, 'abc'), ['a', 'c']);
+  eq(S.splitOnRegex(/d/, 'abc'), ['abc']);
+  eq(S.splitOnRegex(/b(?=c)/, 'abc abd'), ['a', 'c abd']);
+  eq(S.splitOnRegex(/b(?!c)/, 'abc abd'), ['abc a', 'd']);
+
+  eq(S.splitOnRegex(/\s+/, 'foo  bar baz   qux'), ['foo', 'bar', 'baz', 'qux']);
+  eq(S.splitOnRegex(/\s+/, 'foo  bar baz\tqux\nquux'), ['foo', 'bar', 'baz', 'qux', 'quux']);
+  eq(S.splitOnRegex(/\s+/, 'foobar'), ['foobar']);
+  eq(S.splitOnRegex(/;(;)?/, 'foo;;bar;baz'), ['foo', 'bar', 'baz']);
+
+  eq(S.splitOnRegex(/./, 'a\rb\nc'), ['', '\r', '\n', '']);
+  eq(S.splitOnRegex(/[^]/, 'a\nb'), ['', '', '', '']);
+
+  eq(S.splitOnRegex(/^.*$/m, 'foo\nbar'), ['', '\n', '']);
+  eq(S.splitOnRegex(/^.*$/, 'foo\nbar'), ['foo\nbar']);
+
+  eq(S.splitOnRegex(/foo/, 'FOObar'), ['FOObar']);
+  eq(S.splitOnRegex(/foo/i, 'FOObar'), ['', 'bar']);
+
+  eq(S.splitOnRegex(/b/, 'abc'), ['a', 'c']);
+  eq(S.splitOnRegex(/(?:b)/, 'abc'), ['a', 'c']);
```

Comment:
I'm not sure what this is testing, since `/b/` and `/(?:b)/` should be interchangeable. Unless I'm overlooking something, we could remove these two test cases.